cmd/clusterctl/client/cluster/objectgraph.go

```diff
@@ -168,43 +181,63 @@ func (o *objectGraph) objToNode(obj *unstructured.Unstructured) *node {
 		virtual:        false,
 	}
 
+	crdStr := crdTypeMetaToString(metav1.TypeMeta{Kind: newNode.identity.Kind, APIVersion: newNode.identity.APIVersion})
+	if _, ok := o.crds[crdStr]; ok {
+		if o.crds[crdStr].moveLabel {
+			newNode.moveLabel = true
+		}
+	}
+
 	o.uidToNode[newNode.identity.UID] = newNode
 	return newNode
 }
 
 // getDiscoveryTypes returns the list of TypeMeta to be considered for the the move discovery phase.
 // This list includes all the types defines by the CRDs installed by clusterctl and the ConfigMap/Secret core types.
-func (o *objectGraph) getDiscoveryTypes() ([]metav1.TypeMeta, error) {
-	discoveredTypes := []metav1.TypeMeta{}
-
+func (o *objectGraph) getDiscoveryTypes() error {
 	crdList := &apiextensionsv1.CustomResourceDefinitionList{}
 	getDiscoveryTypesBackoff := newReadBackoff()
 	if err := retryWithExponentialBackoff(getDiscoveryTypesBackoff, func() error {
 		return getCRDList(o.proxy, crdList)
 	}); err != nil {
-		return nil, err
+		return err
 	}
 
+	o.crds = make(map[string]*crdInfo)
+
 	for _, crd := range crdList.Items {
 		for _, version := range crd.Spec.Versions {
 			if !version.Storage {
 				continue
 			}
 
-			discoveredTypes = append(discoveredTypes, metav1.TypeMeta{
+			typeMeta := metav1.TypeMeta{
 				Kind: crd.Spec.Names.Kind,
 				APIVersion: metav1.GroupVersion{
 					Group:   crd.Spec.Group,
 					Version: version.Name,
 				}.String(),
-			})
+			}
+
+			o.crds[crdTypeMetaToString(typeMeta)] = &crdInfo{typeMeta: typeMeta}
+			if _, ok := crd.Labels[clusterctlv1.ClusterctlMoveLabelName]; ok {
+				o.crds[crdTypeMetaToString(typeMeta)].moveLabel = true
+			}
+
 		}
 	}
 
-	discoveredTypes = append(discoveredTypes, metav1.TypeMeta{Kind: "Secret", APIVersion: "v1"})
-	discoveredTypes = append(discoveredTypes, metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"})
+	o.crds["secrets.v1"] = &crdInfo{typeMeta: metav1.TypeMeta{Kind: "Secret", APIVersion: "v1"}}
+	o.crds["configmaps.v1"] = &crdInfo{typeMeta: metav1.TypeMeta{Kind: "ConfigMap", APIVersion: "v1"}}
+
+	return nil
+}
 
-	return discoveredTypes, nil
+// TODO: There's gotta be a better way to do this, but I couldn't find an easy way to get the full CRD string
```

Comment:
Really hoping there's a better way to retrieve this info easily.

Reply:
I don't have better ideas, only a suggestion to decouple this from the idea of crdTypeMeta, something like
```suggestion
func getVersionKindString(version, kind string) string {
```